src/plib/controllers/IndexController.php

```diff
@@ -17,7 +17,17 @@ public function init()
 
         $this->client = pm_Session::getClient();
 
-        if ($this->client->isAdmin()) {
+        if ($this->_getParam('dom_id')) {
+            $domainId = $this->_getParam('dom_id');
+
+            if (!$this->client->hasAccessToDomain($domainId)) {
+                throw new Exception('Access denied');
```

Comment:
would be better to use pm_Exception everywhere

Reply:
Fixed